File: chapter04/handling_errors_in_api_endpoints/nanoservices/to_do/core/src/api/basic_actions/create.rs
use crate::enums::TaskStatus;
use crate::structs::ToDoItem;
use dal::json_file::save_one;
use glue::errors::NanoServiceError;


pub fn create(title: &str, status: TaskStatus) -> Result<ToDoItem, NanoServiceError> {
    let item = ToDoItem {
        title: title.to_string(),
        status
    };
    let _ = save_one(&title.to_string(), &item)?;
    Ok(item)
}

File: chapter07/connecting-backend-api-endpoints-to-the-frontend/frontend/src/api/create.ts
import { ToDoItem, ToDoItems, TaskStatus } 
from "../interfaces/toDoItems";
import { postCall } from "./utils";
import { Url } from "./url";


export async function createToDoItemCall(title: string) {
    const toDoItem: ToDoItem = {
        title: title,
        status: TaskStatus.PENDING
    };
    return postCall<ToDoItem, ToDoItems>(
        new Url().create, 
        toDoItem, 
        201
    );
}
